providers/sample/src/java/org/sakaiproject/provider/user/SampleCandidateDetailProvider.java

```diff
@@ -201,7 +202,77 @@ public Optional<List<String>> getAdditionalNotes(User user, Site site){
 		}
 		return Optional.empty();
 	}
-	
+
+	public boolean isAdditionalNotesEnabled(Site site) {
+		try {
+			return (serverConfigurationService.getBoolean(SYSTEM_PROP_DISPLAY_ADDITIONAL_INFORMATION, false) || (site != null && Boolean.parseBoolean(site.getProperties().getProperty(SITE_PROP_DISPLAY_ADDITIONAL_INFORMATION))));
+		} catch(Exception e) {
+			log.warn("Error on isAdditionalNotesEnabled (sample) ", e);
+		}
+		return false;
+	}
+
+	public Optional<List<String>> getSpecialNeeds(User user, Site site){
+		if(site == null) {
+			log.error("getSpecialNeeds: Null site.");
```

Comment:
There is no need to include the method name in the message as it is output autmatically by the logger
```suggestion
			log.warn("A null site was detected, returning empty");
```